lib/web_ui/lib/src/engine/html/recording_canvas.dart

```diff
@@ -515,6 +515,22 @@ class RecordingCanvas {
         drawRRect(rrect, paint);
         return;
       }
+      // Use drawRect for straight line paths painted with a zero strokeWidth
```

Comment:
I'm not sure I actually understand why we want to use drawRect instead of just drawing the path anyway and just changing the stroke width to 1.0. Can you clarify?

Reply:
@eyebrowsoffire
This is related to this part of my PR description: 
"The implementation choice is to draw a filled rectangle with a 1 width or a 1 height. This choice is done for optimization purposes (compared to relying on drawing the line on an Web canvas)."

With the HTML renderer, `drawRect`will be rendered as `<div>`, drawing the path will be rendered as a `<canvas>`. As the use cases for this issue seem to be mainly using a separator on long list, my guess is that it is cheaper to have the browser rendering many <div> insteaf of many <canvas>, but I can be wrong here.

Reply:
No, that's a good call. We want to avoid creating unnecessary canvas elements if a drawing call can be expressed using an HTML element. However, do note that this does not guarantee a div. It depends on the `Paint` and other environmental factors whether a div or a canvas will be used.